pyvista/core/pointset.py

```diff
@@ -48,14 +48,45 @@ def center_of_mass(self, scalars_weight=False):
         alg.Update()
         return np.array(alg.GetCenter())
 
-
     def shallow_copy(self, to_copy):
         """Do a shallow copy the pointset."""
         # Set default points if needed
         if not to_copy.GetPoints():
             to_copy.SetPoints(vtk.vtkPoints())
         return Common.shallow_copy(self, to_copy)
 
+    def remove_cells(self, ind):
```

Comment:
Shouldn't this be a filter? I get that it's not a vtk filter, but semantically it feels like a filter to me (give in a mesh, get out a filtered mesh).

Reply:
It's not quite a filter, but we can change it such that it acts like a filter with an `in-place` method.  Still, I think it should remain a method in `PointSet`.

I'm not actually a fan of `filters.py`, and would prefer that these methods be added directly to the correct class instead.  The filters file is 3000+ LOC, and if we broke up the modules and had one or two classes per module, I think things would be a bit cleaner as we're importing several subclasses for each main class.

But that's a discussion for a later time.

Reply:
I totally agree on filters.py being too large. But for the same reason I like the distinction of methods on classes directly, which deals with loading/saving, accessing data and other properties, and methods in filters, which modify the objects.

Reply:
I'm actually thinking that something like this would be better:
http://www.qtrac.eu/pyclassmulti.html

```python
# _DataStore.py

def bigMethod(state):
    ... # lots of lines

def hugeMethod(state):
    ... # lots of lines
```

```python
# DataStore.py
import _DataStore

class DataStore:

    ... # lots of other methods ...

    def bigMethod(self):
	_DataStore.bigMethod(self)

    def hugeMethod(self):
	_DataStore.hugeMethod(self)
```

In fact, we could even have it so that some of these methods could be called directly from `pyvista`.  This would be more in-line with what `numpy` does.  For examples:
```python
import numpy as np

arr = np.arange(10)
print(arr.max())
print(np.max(arr))
```

We'd have to implement type checking within our functions, but I think that's worth it to be able to call the function directly from `pyvista`.  From there we could run `pyvista.remove_cells`.

If not, I'd rather place methods all within the class or common classes, or place the methods in a separate module and then call them as in above.  This would make for some clean code, especially with `@wraps`.

Reply:
Additionally, I can't really place `remove_cells` in `filters.py` as there isn't a class that  that has methods for just `PolyData` and `UnstructuredGrid`.  Rather, there's a class for all types, `PolyData`, or `UnstructuredGrid`.

I really would prefer to refactor this, and since @banesullivan implemented this, I'll wait until he comes back online to talk about this.  For the time being, I'm leaving this in `PointSet`

Reply:
I'm down to refactor all of that! I created that mess and I could easily reimplement the filters into standalone methods so users could call them directly from a filters submodule (e.g. `from pyvista import filters; filters.contour()`) as well as wrapped methods on classes. This would actually clean up some redefinitions and messiness with how the `MultiBlock` filters are implemented (also, this would be in line with some refactoring/migrating of PVGeo that I want to do... more on that later).

I am very hesitant to make those filters available from the top level of PyVista as that might bloat the API when new users are doing tab-complete. I'd prefer to have a new `filters` module next to the `core` module and force users to import that module much like the `examples` module. That would keep the top level only showing the most relevant API features for creating meshes, plotters, and the few other useful utilities we have available at that level.

```py
import pyvista as pv
from pyvista import filters

mesh = pv.Sphere()

deci = filters.decimate(mesh)
```

This would also make it a bit cleaner for those folks that only want to use PyVista's filters intermixed in their VTK workflows.

Reply:
This refactoring might create a lot of duplicated docs though as the wrapped methods will show in the autodoc of the class and then we probably want to autodoc the whole `filters` module in case folks want to use the standalone methods.